### PR TITLE
Bump to go1.9

### PR DESCRIPTION
**Pull request by caglar10ur**
Release notes: https://golang.org/doc/go1.9

### CODE REVIEW

File: infra/integration-image/Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # docker tag vic-test 192.168.31.15/library/vic-integration-test:1.x
 # docker push 192.168.31.15/library/vic-integration-test:1.x
 
-FROM golang:1.8
+FROM golang:1.9
```
Comment (mhagen-vmware) [inline infra/integration-image/Dockerfile]: You need to rebuild this and upload it, it needs to be vic-integration-test:1.36 currently.  And we need this to go in after #6109 (and make sure you build the image with the changes from 6109)

**Comment by mhagen-vmware** [comment]
Just blocking this so that we can sync the integration image changes, otherwise lgtm

**Comment by mhagen-vmware** [comment]
This should only become a problem if we need to make a ton of changes to our code, then try to cherry pick a later change into 1.2.  So, I don't think we need to wait until after 1.2GA as long as we don't change any other code in order to work with 1.9 (including tests).

**Comment by caglar10ur** [comment]
Also blocked by https://github.com/golang/go/issues/21621

**Comment by caglar10ur** [comment]
We need to bump our number of open files limit during the compilation phase (see ^)

**Comment by vburenin** [comment]
Upgrade to 1.9 before release might be dangerous. You know, most probably people will discover some bad bugs in 1.9, so I better wait for 1.9.1 or 1.9.2
However, for local development purposes we can use 1.9 easy.

**Comment by caglar10ur** [comment]
I think I'll wait till they release 1.9.1 which should be very soon

**Comment by caglar10ur** [comment]
(refreshed the PR with new image)

**Comment by mhagen-vmware** [comment]
Did you build the image and push it according to the instructions at the top of the Dockerfile?

**Comment by caglar10ur** [comment]
wat? no, didn't realized I needed to do that. Why is that step manual?

**Comment by mdubya66** [comment]
should we scrap this and wait for 1.9.1?

**Comment by caglar10ur** [comment]
they already released 1.9.1 but it was a security fix, we need 1.9.2.

**Comment by caglar10ur** [comment]
https://github.com/golang/go/issues/22277 is the tracking issue

**Comment by caglar10ur** [comment]
Now that go 1.9.2 release we can merge this. 

The missing piece is the integration image. Someone with privileges needs to generate that new integration image and push it to local registry.

**Comment by mdubya66** [comment]
@mhagen-vmware let's look at this for 1.3. There are some security fixes in here.

**Comment by mdubya66** [comment]
we changed from 1.9 to 1.9.2

**Comment by mdubya66** [comment]
and a successful build @caglar10ur

**Comment by caglar10ur** [comment]
I believe someone needs to push vic-integration-test image to internal registry either via writing something to automate that or simply running the required steps manually. I prefer that someone not being me as I've no idea how to access those resources.

**Comment by hickeng** [comment]
@mhagen-vmware Looks like 1.10 will be released soon anyway ;)

**Pull request closed without merging**